From dabb0aa2b60ae06f6bbf0adfadf54551adec9b3f Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Wed, 16 Jul 2025 11:03:09 +0000
Subject: [PATCH 002/220] LOCAL: arm64: dts: rockchip: rock5b: disable sdio
 node

The rock-5b board ships with an M2 slot that can be used with PCIe or SDIO
cards so both are described and enabled in device-tree. While there are a
huge number of inexpensive PCIe M2 cards available, there are few SDIO M2
cards in circulation, and none are available from Radxa.

Reflect real-world usage of the 5B board; leave the SDIO node described as
a capability but disable it by default to prevent dmesg errors seen during
mmc probe for an SDIO card that in-practice never exists.

ROCK5B:~ # dmesg | grep mmc2
[    0.790097] mmc_host mmc2: card is non-removable.
[    0.804379] mmc_host mmc2: Bus speed (slot 0) = 400000Hz (slot req 400000Hz, actual 400000HZ div = 0)
[    1.968538] mmc_host mmc2: Timeout sending command (cmd 0x202000 arg 0x0 status 0x80202000)
[    1.992757] mmc_host mmc2: Bus speed (slot 0) = 300000Hz (slot req 300000Hz, actual 300000HZ div = 0)
[    3.163937] mmc_host mmc2: Timeout sending command (cmd 0x202000 arg 0x0 status 0x80202000)
[    3.177872] mmc_host mmc2: Bus speed (slot 0) = 200000Hz (slot req 200000Hz, actual 200000HZ div = 0)
[    4.359405] mmc_host mmc2: Timeout sending command (cmd 0x202000 arg 0x0 status 0x80202000)
[    4.373304] mmc_host mmc2: Bus speed (slot 0) = 187500Hz (slot req 187500Hz, actual 187500HZ div = 0)
[    5.538223] mmc_host mmc2: Timeout sending command (cmd 0x202000 arg 0x0 status 0x80202000)
[    5.539621] mmc2: Failed to initialize a non-removable card

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
index da13dafcbc82..7af53147bdd8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
@@ -27,7 +27,7 @@ &sdio {
 	vqmmc-supply = <&vcc_1v8_s3>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdiom0_pins>;
-	status = "okay";
+	status = "disabled";
 };
 
 &uart6 {
-- 
2.43.0

